packages/gatsby-theme-carbon/src/components/Switcher/Switcher.js

```diff
@@ -1,19 +1,38 @@
-import React, { useContext } from 'react';
+import React, { useContext, useRef, useEffect, useState } from 'react';
 import cx from 'classnames';
 import NavContext from '../../util/context/NavContext';
 import { nav, open, divider, link, linkDisabled } from './Switcher.module.scss';
 
 const Switcher = ({ children }) => {
   const { switcherIsOpen } = useContext(NavContext);
 
+  const listRef = useRef(null);
+  const [height, setHeight] = useState(null);
+
+  // calculate and update height
+  useEffect(() => {
+    switcherIsOpen
+      ? setHeight(listRef.current.offsetHeight + 40)
+      : setHeight(0);
+  }, [listRef, switcherIsOpen]);
+
+  const lgBreakpoint = window.matchMedia('(min-width: 1056px)'); // returns true if screen width matches media query
+  const styles = {
+    container: (isLg, isOpen) => ({
+      /* eslint no-unused-expressions: [2, { allowShortCircuit: true, allowTernary: true }] */
+      maxHeight: !isLg && isOpen ? '100%' : `${height}px`,
+    }),
+  };
+
   return (
     <nav
       className={cx(nav, { [open]: switcherIsOpen })}
       aria-label="IBM Design ecosystem navigation"
       aria-expanded={switcherIsOpen}
       tabIndex="-1"
+      style={styles.container(lgBreakpoint.matches, switcherIsOpen)}
```

Comment:
I think we could clean this up by figuring out what the max-height is supposed to be outside of JSX above and then just passing it in like

```jsx
//outside of jsx
const maxHeight = calculationstuff


// later
style={{ maxHeight }}
```